enhancements/windows-containers/health-management.md

```diff
@@ -0,0 +1,311 @@
+Windows Instance Config Daemon (WICD). WICD will have the following responsibilities:
+* WICD maintains current responsibilities of bootstrapping kubelet through the `bootstrap` command.
+* Using the configuration provided by a ConfigMap created by WMCO, WICD maintains the state of Windows Services on the instance.
+* WICD reverts all changes made to an instance when ran with the `cleanup` command, and also deletes the Node
+  object.
+
+To enable this, WMCO will have these changes in responsibility:
+* WMCO will no longer configure the Windows services on Windows instances, except for WICD.
+* WMCO will create and maintain a ConfigMap which provides WICD with the specifications for each Windows service that
+  must be created on a Windows instance.
+* WMCO will invoke the `bootstrap` command of WICD when initially adding a Windows instance to the cluster.
+* WMCO will invoke the `cleanup` command of WICD when removing an instance from the cluster. WMCO will no longer
+  delete the Node object.
+
+The Windows Node creation workflow for the end user will not be changed.
+
+### User Stories
+
+User stories can be found within the health management epic, [WINC-657](https://issues.redhat.com/browse/WINC-657).
+
+### API Extensions
+
+N/A
+
+### Risks and Mitigations
+
+* It is possible that issues with Windows instances will occur which WICD is unable to resolve, causing the Node to
+  enter a `NotReady` state. For instances created through the machine-api, the user must set up MachineHealthChecks,
+  which will correct the state by recreating the Machine. For BYOH instances, user intervention is required at this
+  point. Events and alerts will ensure that a cluster administrator is made aware of the issue.
+* There is a possibility that a cluster admin could mistakenly remove the services ConfigMap for an older WMCO version,
+  before Node upgrade is completed. This would result in WICD not knowing what services it has configured on the instance.
+  In order to prevent this, WICD can set the Windows service description to "openshift-managed", so that it
+  can search for services and remove them. In order to facilitate upgrades from a version without WICD, this string
+  should be added to all existing services in 4.10, one major version ahead of WICD's target release.
+
+## Design Details
+
+### Services ConfigMap
+
+WMCO will create and maintain an immutable ConfigMap with the naming scheme:
+`services-<MajorVersion>-<MinorVersion>-<PatchVersion>-<Commit>`. For example, if WMCO was built from commit a7b5
+of WMCO version 5.0.0, the created ConfigMap would be named `services-5-0-0-a7b5`.
+
+The purpose of this ConfigMap is to provide WICD with the specifications required to configure a Node. The ConfigMap
+will be watched via the existing ConfigMap controller. WMCO is the owner of the ConfigMap and the source of the data
+present within it. If an entity attempts to create a services ConfigMap with incorrect values, WMCO will delete it.
```

Comment:
What I don't understand is how would there be a ConfigMap with incorrect values given that to create such an incorrect ConfigMap it would have to be deleted first?

Reply:
Thank you for the explanation. This bit of detail will be useful to have in the enhancement.